File: src/use-cases/implementations/forgot-password.ts
import { inject, injectable } from "inversify";
import { Repositories } from "../../common/dependency-injection";
import { IAuthRepository } from "../../repositories/interfaces";
import { IForgotPasswordUseCase } from "../interfaces";

@injectable()
export class ForgotPasswordUseCase implements IForgotPasswordUseCase {
  authRepo: IAuthRepository;

  constructor(@inject(Repositories.auth) authRepo: IAuthRepository) {
    this.authRepo = authRepo;
  }

  async execute(email: string) {
    if (!(await this.authRepo.forgotPassword(email))) {
      throw new Error("Could not send forgot password email");
    }
  }
}

File: lib/src/use-cases/implementations/forgot_password.dart
import '../../entities/entities.dart';
import '../../repositories/repositories.dart';
import '../interfaces/interfaces.dart';

class ForgotPasswordUseCase implements IForgotPasswordUseCase {
  @override
  Future<Failure?> forgotPassword(String email) =>
      authRepo.forgotPassword(email);

  @override
  Future<Failure?> resetPassword(
    String email,
    String code,
    String newPassword,
  ) =>
      authRepo.resetPassword(email, code, newPassword);
}
